### ISSUE TITLE
tf.estimator.train_and_evaluation stops before it reaches max_step

### ISSUE THREAD

**Issue by litaxc** [tensorflow/tensorflow#23248]
**System information**
- TensorFlow version: 1.11.0, using docker images tensorflow/tensorflow:1.11.0-gpu-py3
- Doc Link: [tf.estimator.train_and_evaluation](https://www.tensorflow.org/api_docs/python/tf/estimator/train_and_evaluate)

**Describe the documentation issue**

According to the doc, `tf.estimator.train_and_evaluation` was meant to train forever or `max_step` if specified in `train_spec`.

However, it took me many trial and errors to realize: if the dataset returned by `input_fn` in the `train_spec` was repeated a finite number of times and that number was smaller than `max_step`, the training would stop before it reached `max_step`.

For example, I had a dataset about 120000 images. With batch size of 64, the training would stop at 1875 step (~120000/64), even though `max_step` was set at 10000 or higher.

This behavior is not written in the document. Should the document be update or was it an unexpected behavior of the function?

**We welcome contributions by users. Will you be able to update submit a PR (use the [doc style guide](https://www.tensorflow.org/community/documentation)) to fix the doc Issue?**

Yes.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ymodak: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by lasclocker** [comment]
You should not limit repeat num, please use `repeat()`, and it will repeat the input indefinitely.
